src/handlers/cors.js

```diff
@@ -1,17 +1,79 @@
-module.exports = function corsHandler({ allowedOrigins = ['*'] }) {
+module.exports = function corsHandler(
+  {
+    allowedOrigins = ['*'],
+    allowedMethods = ['GET','PUT','POST','PATCH','DELETE','HEAD','OPTIONS'],
+    allowCredentials = true,
+    allowedHeaders = ['Content-Type'],
+    allowedExposeHeaders = ['WWW-Authenticate', 'Server-Authorization'],
+    maxAge = '600',
+    optionsSuccessStatus = 204,
+    terminatePreflight = false
+  }) {
   return async (ctx, next) => {
-    const { origin } = ctx.request.headers;
-
-    if (origin) {
-      if (allowedOrigins[0] === '*' || allowedOrigins.indexOf(origin) !== -1) {
-        ctx.set('Access-Control-Allow-Credentials', 'true');
-        ctx.set('Access-Control-Allow-Origin', origin);
-        ctx.set('Access-Control-Allow-Methods', 'GET,PUT,POST,PATCH,DELETE,HEAD,OPTIONS');
-        ctx.set('Access-Control-Allow-Headers', 'Content-Type');
-        ctx.set('Access-Control-Expose-Headers', 'WWW-Authenticate,Server-Authorization');
+    const method = ctx.request.method
+    const origin = ctx.request.headers.origin
+    const requestHeaders = ctx.request.headers['access-control-request-headers']
+
+
+    if (method === 'OPTIONS') {
+      configureOrigin(ctx, origin, allowedOrigins);
+      configureCredentials(ctx, allowCredentials);
+      configureMethods(ctx, allowedMethods);
+      configureAllowedHeaders(ctx, requestHeaders, allowedHeaders);
+      configureMaxAge(ctx, maxAge);
+      configureExposedHeaders(ctx, allowedExposeHeaders);
+      if (terminatePreflight) {
+        ctx.status = optionsSuccessStatus;
+        ctx.set('Content-Length', '0');
```

Comment:
Is this a documented behavior? Is there any harm in setting it so it would be explicit.
My implementation is heavily influenced by express cors plugin: https://github.com/expressjs/cors/blob/master/lib/index.js

There's a comment there that says that that safari and probably other browsers will hang on 204 response if there's no explicit content-length header set to 0.

Reply:
There's no harm in setting it but pretty certain that it will be replaced any way.

Reply:
I prefer leaving this so the intent is clear, even if cloudflare change their defaults.